lib/mix/lib/mix.ex

```diff
@@ -823,6 +834,13 @@ defmodule Mix do
     end
   end
 
+  defp expand_path(_path = nil, _deps, _relative_path), do: nil
+  defp expand_path(path, _deps, _relative_path) when is_binary(path), do: path
+
+  defp expand_path(app_name, deps, relative_path) when is_atom(app_name) do
+    deps |> Keyword.fetch!(app_name) |> Keyword.fetch!(:path) |> Path.join(relative_path)
```

Comment:
Unfortunately path dependencies can be written in the format:

```elixir
{:dep, "~> 0.1.0", path: ...}
```

My suggestion would be something like this:

```elixir
case List.keyfind(deps, app_name, 0) do
  {_, _, opts} -> opts[:path] || raise ArgumentError, "#{app_name} given to :config_path must be a path dependency"
  {_, opts} -> opts[:path] || raise ArgumentError, "#{app_name} given to :config_path must be a path dependency"
  _ ->  raise ArgumentError, "unknown dependency #{app_name} given to :config_path"
end
```

Reply:
Oh sorry, I totally missed that 👍
Will fix.

Reply:
Done in [c23bdda](https://github.com/elixir-lang/elixir/pull/12204/commits/c23bdda44c740b080fa45cda9b8ab43a1eda1a79)